plugins/faustwp/includes/replacement/callbacks.php

```diff
@@ -31,36 +31,42 @@
  *
  * @return string The post content.
  */
-function content_replacement( $content ) {
-	$use_wp_domain_for_permalinks = ! domain_replacement_enabled();
-	$use_wp_domain_for_media      = use_wp_domain_for_media();
+function content_replacement( string $content ): string {
 
-	if ( $use_wp_domain_for_permalinks && $use_wp_domain_for_media ) {
+	if ( ! $content ) {
+		return '';
+	}
+
+	$replace_content_urls = domain_replacement_enabled();
+	$replace_media_urls   = ! use_wp_domain_for_media();
+
+	if ( ! $replace_content_urls && ! $replace_media_urls ) {
 		return $content;
 	}
 
-	$replacement = faustwp_get_setting( 'frontend_uri' );
-	if ( ! $replacement ) {
-		$replacement = '/';
+	$wp_site_urls = faustwp_get_wp_site_urls();
+	if ( empty( $wp_site_urls ) ) {
+		return $content;
 	}
 
-	$site_url  = site_url();
-	$media_dir = str_replace( $site_url, '', wp_upload_dir()['baseurl'] );
-	$media_url = $site_url . $media_dir;
+	$wp_media_urls       = faustwp_get_wp_media_urls();
+	$relative_upload_url = faustwp_get_relative_upload_url( $wp_site_urls );
+	$frontend_uri        = faustwp_get_setting( 'frontend_uri' );
+	if ( ! $frontend_uri ) {
+		$frontend_uri = '/';
+	}
 
-	if ( $use_wp_domain_for_permalinks && ! $use_wp_domain_for_media ) {
-		$content = str_replace( $media_url, $replacement . $media_dir, $content );
-		return $content;
+	if ( $replace_content_urls && $replace_media_urls ) {
```

Comment:
Actually, if run the portion of the code for the `! $replace_media_urls ` first we can save more space here.

```suggestion
    if ( ! $replace_media_urls ) {
		$site_urls_pattern = implode( '|', array_map( 'preg_quote', $wp_site_urls ) );
		$pattern           = '#(' . $site_urls_pattern . ')(?!' . $relative_upload_url . '(\/|$))#';

		return preg_replace( $pattern, $frontend_uri, $content );
	}

	$replace = $frontend_uri . ( ! $replace_content_urls ? $relative_upload_url : '' );

	return str_replace( $wp_media_urls, $replace, $content );
```

Reply:
I tested this locally and it fails a few tests. The issue is we have 3 conditions here (content and media = true, content = true or media = true).

I will see if I can take this approach and get it to work 😄

Reply:
Updated the PR. It should be more readable now but that part was added at the bottom of that method.